scripts/gas-refund-program/computeGasRefund.ts

```diff
@@ -13,64 +13,77 @@ import { GasRefundParticipation } from '../../src/models/GasRefundParticipation'
 import { init, resolveEpochCalcTimeInterval } from './common';
 import { EpochInfo } from '../../src/lib/epoch-info';
 import { CHAIN_ID_MAINNET } from '../../src/lib/constants';
+import Database from '../../src/database';
 
 const logger = global.LOGGER('GRP');
 
 async function startComputingGasRefundAllChains() {
-  await init({ epochPolling: true });
+  await init({
+    epochPolling: true,
+    dbTransactionNamespace: 'gas-refund-computation',
+  });
 
   const epochInfo = EpochInfo.getInstance(CHAIN_ID_MAINNET, true);
 
-  return Promise.allSettled(
-    GRP_SUPPORTED_CHAINS.map(async chainId => {
-      const lastEpochProcessed = await GasRefundParticipation.max<
-        number,
-        GasRefundParticipation
-      >('epoch', {
-        where: {
-          isCompleted: false,
-          chainId,
-        },
-      });
+  return Database.sequelize.transaction(async () => {
+    await Database.sequelize.query(
+      `LOCK TABLE "${GasRefundParticipation.tableName}" IN ACCESS EXCLUSIVE MODE;`,
+    );
```

Comment:
This transaction is applied to every sequilize query in the call stack that starts here?
Did you consider starting separate transactions for each `GRP_SUPPORTED_CHAINS.map(...` instead or do you want _all or nothing_ specifically?

Reply:
I considered it, though cls-hooked doesn't allow it (all or nothing comes from it).
If you think it's really worth segmenting by chainIds then we'd have to pass transaction deeply which is was I tried to avoid with cls.